lib/writers/git.go

```diff
@@ -406,3 +498,83 @@ func sshUsernameFromURL(url string) (string, error) {
 	}
 	return ep.User, nil
 }
+
+// for unit tests
+
+var GenerateGitHubAppJWT = generateGitHubAppJWT
+var GetGitHubInstallationToken = getGitHubInstallationToken
+
+// generateGitHubAppJWT creates a signed JWT for GitHub App authentication
+func generateGitHubAppJWT(appID string, privateKey string) (string, error) {
+	block, _ := pem.Decode([]byte(privateKey))
+	if block == nil {
+		return "", errors.New("invalid private key: failed to parse PEM block")
+	}
+
+	var parsedKey any
+	var err error
+
+	if key, err := x509.ParsePKCS1PrivateKey(block.Bytes); err == nil {
+		parsedKey = key
+	} else if key, err := x509.ParsePKCS8PrivateKey(block.Bytes); err == nil {
+		if rsaKey, ok := key.(*rsa.PrivateKey); ok {
+			parsedKey = rsaKey
+		} else {
+			return "", errors.New("private key is not RSA")
+		}
+	} else {
+		return "", fmt.Errorf("failed to parse private key: %w", err)
+	}
```

Comment:
change: this is very hard to read and not fully tested 👀 (i commented out the inner parsing and it still passed all the tests)

A simple helper function can make this much more readable

```go
func parseRSAKey(block *pem.Block) (*rsa.PrivateKey, error) {
	if k, err := x509.ParsePKCS1PrivateKey(block.Bytes); err == nil {
		return k, nil
	}

	k, err := x509.ParsePKCS8PrivateKey(block.Bytes)
	if err != nil {
		return nil, errors.New("failed to parse RSA private key")
	}
	if rsaKey, ok := k.(*rsa.PrivateKey); ok {
		return rsaKey, nil
	}
	return nil, errors.New("private key is not RSA")
}
```